x/ccv/types/ccv.go

```diff
@@ -99,7 +120,45 @@ func (cp ConsumerPacketData) ValidateBasic() (err error) {
 	return
 }
 
+// Convert to bytes while maintaining over the wire compatibility with previous versions.
 func (cp ConsumerPacketData) GetBytes() []byte {
-	bytes := ModuleCdc.MustMarshalJSON(&cp)
+	return cp.ToV1Bytes()
+}
+
+// ToV1Bytes converts the ConsumerPacketData to JSON byte array compatible
+// with the format used by ICS versions using cosmos-sdk v45 (ICS v1 and ICS v2).
+func (cp ConsumerPacketData) ToV1Bytes() []byte {
```

Comment:
Important to check and inspired by Shawn's solution from https://github.com/cosmos/interchain-security/pull/1089